include/mapbox/vector_tile.hpp

```diff
@@ -152,48 +147,54 @@ optional<mapbox::geometry::value> feature::getValue(const std::string& key) cons
         return optional<mapbox::geometry::value>();
     }
 
+    const auto values_count = layer_.values.size();
+    const auto keymap_count = layer_.keysMap.size();
     auto start_itr = tags_iter.begin();
     const auto end_itr = tags_iter.end();
     while (start_itr != end_itr) {
-        uint32_t tag_key = static_cast<uint32_t>(*start_itr++);
+        std::uint32_t tag_key = static_cast<std::uint32_t>(*start_itr++);
 
-        if (layer_.keysMap.size() <= tag_key) {
+        if (keymap_count <= tag_key) {
```

Comment:
@flippmoke my reading is that this helps perf because otherwise `layer_.keysMap.size()` is called for every pass around the `while` loop which could be multiple times.